bloomfilter/src/Data/BloomFilter/Blocked/BitArray.hs

```diff
@@ -34,7 +34,7 @@ import           Data.Primitive.PrimArray
 import           Data.Word (Word64, Word8)
 
 import           GHC.Exts (Int (I#), prefetchByteArray0#,
-                     prefetchMutableByteArray3#)
+                     prefetchMutableByteArray0#)
```

Comment:
What does prefetch 0 vs. prefetch 3 mean again? I read up on `__builtin_prefetch`, which seemed to be the inspiration for the primop in GHC, and there it means temporal locality. Does temporal locality relate directly to cache levels?

Reply:
Values 1--3 mean how many levels of cache to load it into: 1 level (L3 only), 2 levels (L3+L2), 3 levels (L3+L2+L1).

Value 0 means, as you say, means prefetch with low temporal locality. This means read it into some level of cache (unspecified which ones) but mark it with a hint that it will not be needed for long. In the CPU I think this means that the entry is marked so that it is the first to be evicted when that cache entry is next needed, so will not hang around for long or compete with other data that needs to be kept in the cache.

Reply:
Documented.